### PR TITLE
Fixes a rare interaction between mice by assigning a proper subtype.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
...Oh fuck, we left Jerry on donut.

## About The Pull Request

Cats named Jerry can't catch mice named Tom. This is known. However, there was only one distinct instance of jerry, and that was on DonutStation in the permabrig. ...And we don't have donutstation in the rotation anymore, so this interaction was pretty rare and now basically impossible to pull off.

This PR finally makes Jerry his own subtype, and throws the ability to catch mice onto an `inept_hunter` var, so it can be varedited or adjusted in the future.

## Why It's Good For The Game

Effectively a tiny code clean up on a meme feature that might see some use now that it's now literally the most obscure ever.

## Changelog
:cl:
add: Jerry the cat is now his own subtype, and cannot catch Tom the Mouse.
/:cl:

### CODE REVIEW

**Comment by Winterous** [comment]
Isn't this the wrong way around? Tom was the cat.

**Comment by MrPerson** [comment]
Shouldn't Tom have an artful_dodger type var if Jerry is gonna have an inept_hunter var?

**Comment by ArcaneMusic** [comment]
> Isn't this the wrong way around? Tom was the cat.

Someone long before me made the switch, no sense in changing it back now 🤷

**Comment by Tharcoonvagh** [comment]
> Isn't this the wrong way around? Tom was the cat.

That's the joke

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -261,6 +263,11 @@
 				stop_automated_movement = 1
 				walk_to(src,movement_target,0,3)
 
+/mob/living/simple_animal/pet/cat/jerry //Holy shit we left jerry on pubby ~ Arcane
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:266]:
```suggestion
/mob/living/simple_animal/pet/cat/jerry //Holy shit we left jerry on donut ~ Arcane ~Fikou
```

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -221,7 +223,7 @@
 	if((src.loc) && isturf(src.loc))
 		if(!stat && !resting && !buckled)
 			for(var/mob/living/simple_animal/mouse/M in view(1,src))
-				if(istype(M, /mob/living/simple_animal/mouse/brown/tom) && (name == "Jerry")) //Turns out there's no jerry subtype.
+				if(istype(M, /mob/living/simple_animal/mouse/brown/tom) && inept_hunter) //We have a jerry subtype now.
```
Comment (Jared-Fogle) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:226]:
```suggestion
				if(istype(M, /mob/living/simple_animal/mouse/brown/tom) && inept_hunter)
```

This comment makes no sense without context, and so isn't necessary.

**Pull request merged**